profiler/profiler_test.go

```diff
@@ -334,3 +339,72 @@ func unstartedProfiler(opts ...Option) (*profiler, error) {
 	p.uploadFunc = func(_ batch) error { return nil }
 	return p, nil
 }
+
+func TestAllUploaded(t *testing.T) {
```

Comment:
I like smoke tests like this 👍. Much nicer than the other tests that rely on hacking up internals to intercept calls and stuff.